services/order.py

```diff
@@ -0,0 +1,54 @@
+from datetime import datetime
+
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import transaction
+
+
+from db.models import Order, Ticket
+
+
+@transaction.atomic()
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: datetime = None) -> Order:
+
+    try:
+        user = get_user_model().objects.get(username=username)
+    except ObjectDoesNotExist:
+        raise ValueError(f"User with username '{username}' does not exist.")
+
+    order = Order.objects.create(user=user)
+
+    if date:
+        order.created_at = date
+        order.save()
```

Comment:
The `save()` method is called twice in the `create_order` function: once after setting the `created_at` date and again implicitly when tickets are associated with the order. Consider refactoring to call `save()` only once after all modifications are complete to avoid redundancy.